content/gocourse-2019-05-melbourne.slide

```diff
@@ -1003,6 +1003,46 @@ Use `gofmt` or `goimport` as part of your editor/IDE setup
 - ...
 
 
+* JSON - Marshal / Unmarshal
+
+- Encoding a struct or primitive converts it into a json encoded stream
+- Marshalling uses an encoder to convert a struct / primitive into an array of json bytes, which are stored in memory
+- Decoding / Unmarshalling are the opposing functions to encoding and marshalling
+
+	func Marshal(v interface{}) ([]byte, error) {
+
```

Comment:
Probably remove function body (missing return values) - just print function signature.

Reply:
I went down this approach originally but the colouring of the slides was slightly off (The Unmarshal function didn't have the correct colour syntax)

Do you still want me to go down this path?

Reply:
Ok so I have removed this, it does result in an inconsistency with the colouring of the two funcs but I will let you decide whether it is okay or not.

Reply:
Yeah I like that one better. Updated